The ST M28W320FSx and M28W640FSx FLASH devices do not support
the Intel extension buffer write mode.  They only have a double
and quad word write extension.

This patch disables buffered writes for these devices.

Signed-off-by: Carl Shaw <carl.shaw@st.com>
Index: linux/drivers/mtd/chips/cfi_cmdset_0001.c
===================================================================
--- linux.orig/drivers/mtd/chips/cfi_cmdset_0001.c
+++ linux/drivers/mtd/chips/cfi_cmdset_0001.c
@@ -181,11 +181,12 @@ static void fixup_no_write_suspend(struc
 }
 #endif
 
-static void fixup_st_m28w320ct(struct mtd_info *mtd, void* param)
+static void fixup_st_m28wXX0_disable_bufferwrite(struct mtd_info *mtd, void* param)
 {
 	struct map_info *map = mtd->priv;
 	struct cfi_private *cfi = map->fldrv_priv;
 
+	printk(KERN_INFO "Using word write for ST M28WXX0 FLASH\n");
 	cfi->cfiq->BufWriteTimeoutTyp = 0;	/* Not supported */
 	cfi->cfiq->BufWriteTimeoutMax = 0;	/* Not supported */
 }
@@ -227,11 +228,17 @@ static struct cfi_fixup cfi_fixup_table[
 #ifdef CMDSET0001_DISABLE_WRITE_SUSPEND
 	{ CFI_MFR_ANY, CFI_ID_ANY, fixup_no_write_suspend, NULL },
 #endif
+	{ CFI_MFR_ST, 0x00ba, /* M28W320CT */ fixup_st_m28wXX0_disable_bufferwrite, NULL },
+	{ CFI_MFR_ST, 0x00bb, /* M28W320CB */ fixup_st_m28w320cb, NULL },
+	{ CFI_MFR_ST, 0x8857, fixup_st_m28wXX0_disable_bufferwrite, NULL },
+	{ CFI_MFR_ST, 0x8858, fixup_st_m28wXX0_disable_bufferwrite, NULL },
+	{ CFI_MFR_ST, 0x8859, fixup_st_m28wXX0_disable_bufferwrite, NULL },
+	{ CFI_MFR_ST, 0x880a, fixup_st_m28wXX0_disable_bufferwrite, NULL },
+	{ CFI_MFR_ST, 0x880b, fixup_st_m28wXX0_disable_bufferwrite, NULL },
+	{ CFI_MFR_ST, 0x880c, fixup_st_m28wXX0_disable_bufferwrite, NULL },
 #if !FORCE_WORD_WRITE
 	{ CFI_MFR_ANY, CFI_ID_ANY, fixup_use_write_buffers, NULL },
 #endif
-	{ CFI_MFR_ST, 0x00ba, /* M28W320CT */ fixup_st_m28w320ct, NULL },
-	{ CFI_MFR_ST, 0x00bb, /* M28W320CB */ fixup_st_m28w320cb, NULL },
 	{ 0, 0, NULL, NULL }
 };
 
